src/core/packet_builder.c

```diff
@@ -68,7 +68,6 @@ QuicPacketBuilderValidate(
         CXPLAT_DBG_ASSERT(Builder->Datagram->Length != 0);
         CXPLAT_DBG_ASSERT(Builder->Datagram->Length <= UINT16_MAX);
         CXPLAT_DBG_ASSERT(Builder->Datagram->Length >= Builder->MinimumDatagramLength);
-        CXPLAT_DBG_ASSERT(Builder->Datagram->Length > (uint32_t)Builder->DatagramLength);
```

Comment:
I'm still not sure about this.

Reply:
Well its strictly worse then the check right below this. If the below check passes, this one is guaranteed to pass as well (As long as EncryptionOverhead is not somehow negative or 0 and the compiler does signed math).